File: src/main/java/chat/teco/tecochat/chat/domain/chat/event/ChatCreatedEvent.java
package chat.teco.tecochat.chat.domain.chat.event;

import chat.teco.tecochat.chat.domain.chat.Chat;
import chat.teco.tecochat.common.event.BaseEvent;
import chat.teco.tecochat.common.event.BaseEventHistory;
import java.time.LocalDateTime;

public class ChatCreatedEvent extends BaseEvent {

    private final Long chatId;
    private final LocalDateTime localDateTime;

    public ChatCreatedEvent(
            Long chatId,
            LocalDateTime localDateTime
    ) {
        this.chatId = chatId;
        this.localDateTime = localDateTime;
    }

    public static ChatCreatedEvent from(Chat chat) {
        return new ChatCreatedEvent(chat.id(), LocalDateTime.now());
    }

    public Long chatId() {
        return chatId;
    }

    public LocalDateTime localDateTime() {
        return localDateTime;
    }

    @Override
    public BaseEventHistory history() {
        return new ChatCreatedEventHistory(localDateTime, chatId);
    }
}

File: src/main/kotlin/chat/teco/tecochat/domain/chat/ChatCreatedEvent.kt
package chat.teco.tecochat.domain.chat

import chat.teco.tecochat.support.domain.BaseEvent
import chat.teco.tecochat.support.domain.BaseEventHistory
import java.time.LocalDateTime

class ChatCreatedEvent(
    val chatId: Long,
    val localDateTime: LocalDateTime,
) : BaseEvent() {

    override fun history(): BaseEventHistory {
        return ChatCreatedEventHistory(localDateTime, chatId)
    }

    companion object {
        @JvmStatic
        fun from(chat: Chat): ChatCreatedEvent {
            return ChatCreatedEvent(chat.id, LocalDateTime.now())
        }
    }
}
